### PR TITLE
[4.3] Use the internal database in the sysinfo model

### PR DESCRIPTION
**Pull request by laoneo**
### Summary of Changes
The system information model should use the internal database.

### Testing Instructions
- Open the link /administrator/index.php?option=com_admin&view=sysinfo
- Download the config as JSON

### Actual result BEFORE applying this Pull Request
The JSON file should contain all extensions.

### Expected result AFTER applying this Pull Request
The JSON file should contain all extensions.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: administrator/components/com_admin/src/Model/SysinfoModel.php [hunk 1]
```diff
@@ -436,8 +436,8 @@ public function getPhpInfoArray(): array
     public function getExtensions(): array
     {
         $installed = [];
-        $db = Factory::getContainer()->get('DatabaseDriver');
-        $query = $db->getQuery(true)
+        $db        = $this->getDatabase();
```
Comment (wilsonge) [inline administrator/components/com_admin/src/Model/SysinfoModel.php:439]: I think we should use getDbo() to be consistent with all the other models in core where we chose to keep the legacy method until 5.0. Obviously moving away from the static container is a good thing though

Comment (laoneo) [reply administrator/components/com_admin/src/Model/SysinfoModel.php:439]: Extension models can safely use the new function, the libraries one not.

**Review by wilsonge — APPROVED** [review]

**Comment by sandewt** [comment]
I have tested this item :white_check_mark: successfully on 92cb34273da978f81f4ae5e4ae4904fe44a3479b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38995">issues.joomla.org/tracker/joomla-cms/38995</a>.</sub>

**Comment by RickR2H** [comment]
I have tested this item :white_check_mark: successfully on 92cb34273da978f81f4ae5e4ae4904fe44a3479b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38995">issues.joomla.org/tracker/joomla-cms/38995</a>.</sub>

**Comment by RickR2H** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38995">issues.joomla.org/tracker/joomla-cms/38995</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**